From d59c392314c864363647dc281358cc5acbc1ca22 Mon Sep 17 00:00:00 2001
From: Alexander Forsch <alexander.forsch@bytesatwork.ch>
Date: Tue, 22 Jul 2025 10:29:55 +0200
Subject: [PATCH] drivers: gpu/drm: Adapt for byteENGINE IMX8MM

Disable video burst mode in Youritech ILI9806E panel
for byteENGINE IMX8MM.

Signed-off-by: Alexander Forsch <alexander.forsch@bytesatwork.ch>
---
 drivers/gpu/drm/panel/Kconfig                    | 9 +++++++++
 drivers/gpu/drm/panel/panel-youritech-ili9806e.c | 3 +++
 2 files changed, 12 insertions(+)

diff --git a/drivers/gpu/drm/panel/Kconfig b/drivers/gpu/drm/panel/Kconfig
index a135a1b2ca..a3eaf6c187 100644
--- a/drivers/gpu/drm/panel/Kconfig
+++ b/drivers/gpu/drm/panel/Kconfig
@@ -880,4 +880,13 @@ config DRM_PANEL_YOURITECH_ILI9806E
 	help
 	  Say Y here if you want to enable support for Youritech
 	  ILI9806E DSI panel.
+
+config DRM_PANEL_YOURITECH_ILI9806E_DISABLE_BURST_MODE
+	bool "Disable video burst mode"
+	depends on DRM_PANEL_YOURITECH_ILI9806E
+	default n
+	help
+	  Disable video burst mode for the Youritech ILI9806E panel.
+	  This is required for some panel/SoC combinations to work correctly.
+	  If unsure, say N.
 endmenu
diff --git a/drivers/gpu/drm/panel/panel-youritech-ili9806e.c b/drivers/gpu/drm/panel/panel-youritech-ili9806e.c
index 7450b6027f..8e2c3cbac8 100644
--- a/drivers/gpu/drm/panel/panel-youritech-ili9806e.c
+++ b/drivers/gpu/drm/panel/panel-youritech-ili9806e.c
@@ -490,6 +490,9 @@ static int ili9806e_panel_probe(struct mipi_dsi_device *dsi)
 	dsi->format = MIPI_DSI_FMT_RGB888;
 	dsi->mode_flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST |
 			  MIPI_DSI_MODE_LPM;
+#ifdef CONFIG_DRM_PANEL_YOURITECH_ILI9806E_DISABLE_BURST_MODE
+	dsi->mode_flags &= ~MIPI_DSI_MODE_VIDEO_BURST;
+#endif
 	ret = of_property_read_u32(np, "video-mode", &video_mode);
 	if (!ret) {
 		switch (video_mode) {
